File: samples/VisualBasic/Northwind/Supplier.vb
Imports System
Imports System.Collections.ObjectModel
Imports DbExtensions

Namespace Northwind

   <Table(Name:="Suppliers")>
   Public Class Supplier

      <Column>
      Property Address As String

      <Column>
      Property City As String

      <Column>
      Property CompanyName As String

      <Column>
      Property ContactName As String

      <Column>
      Property ContactTitle As String

      <Column>
      Property Country As String

      <Column>
      Property Fax As String

      <Column>
      Property Phone As String

      <Column>
      Property PostalCode As String

      <Column>
      Property [Region] As String

      <Column(IsPrimaryKey:=True, IsDbGenerated:=True)>
      Property SupplierID As Integer

      <Association(OtherKey:=NameOf(Product.SupplierID))>
      ReadOnly Property Products As New Collection(Of Product)

   End Class

End Namespace

File: samples/CSharp/Northwind/Supplier.cs
using System;
using System.Collections.ObjectModel;
using DbExtensions;

namespace Samples.CSharp.Northwind {

   [Table(Name = "Suppliers")]
   public class Supplier {

      [Column(IsPrimaryKey = true, IsDbGenerated = true)]
      public int SupplierID { get; set; }

      [Column]
      public string CompanyName { get; set; }

      [Column]
      public string ContactName { get; set; }

      [Column]
      public string ContactTitle { get; set; }

      [Column]
      public string Address { get; set; }

      [Column]
      public string City { get; set; }

      [Column]
      public string Region { get; set; }

      [Column]
      public string PostalCode { get; set; }

      [Column]
      public string Country { get; set; }

      [Column]
      public string Phone { get; set; }

      [Column]
      public string Fax { get; set; }

      [Association(OtherKey = nameof(Product.SupplierID))]
      public Collection<Product> Products { get; } = new Collection<Product>();
   }
}
